third_party/move/move-model/src/builder/exp_builder.rs

```diff
@@ -3722,6 +3729,72 @@ impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'mo
         }
     }
 
+    fn translate_test(
+        &mut self,
+        loc: &Loc,
+        exp: &EA::Exp,
+        tys: &[EA::Type],
+        expected_type: &Type,
+        context: &ErrorMessageContext,
+    ) -> ExpData {
+        let (exp_ty, exp) = self.translate_exp_free(exp);
+        let mut variants = vec![];
+        let mut struct_id = None;
+        for ty in tys {
+            let loc = self.to_loc(&ty.loc);
+            if let EA::Type_::Apply(maccess, generics) = &ty.value {
+                let pat = self.translate_lvalue_unpack(
```

Comment:
Can we rename this method, since we're using it for things other than `unpack` here?  Perhaps you can put shared functionality into a new method called by both `translate_lvalue_unpack` and `translate_test`, so you don't have to have the awkward boolean parameter to select which one is being run.

Reply:
Good point. I was to lazy to do it at first, its a bit of code, now done and pulled out `translate_constructor_name`. Note that this also lead to some error message changes elsewhere because a previous inference error location bug is gone.

Reply:
[Graphie test]